quidem-it/src/main/java/org/apache/druid/quidem/ExposedAsBrokerQueryComponentSupplierWrapper.java

```diff
@@ -0,0 +1,371 @@
+    @LazySingleton
+    public BrokerSegmentMetadataCache provideCache()
+    {
+      return null;
+    }
+
+    @Provides
+    @LazySingleton
+    public Properties getProps()
+    {
+      Properties localProps = new Properties();
+      localProps.put("druid.enableTlsPort", "false");
+      localProps.put("druid.zk.service.enabled", "false");
+      localProps.put("druid.plaintextPort", "12345");
+      localProps.put("druid.host", "localhost");
+      localProps.put("druid.broker.segment.awaitInitializationOnStart", "false");
+      return localProps;
+    }
+
+    @Provides
+    @LazySingleton
+    public SqlEngine createMockSqlEngine(
+        final QuerySegmentWalker walker,
+        final QueryRunnerFactoryConglomerate conglomerate,
+        @Json ObjectMapper jsonMapper)
+    {
+      return new NativeSqlEngine(CalciteTests.createMockQueryLifecycleFactory(walker, conglomerate), jsonMapper);
+    }
+
+    @Provides
+    @LazySingleton
+    DruidNodeDiscoveryProvider getDruidNodeDiscoveryProvider()
+    {
+      final DruidNode coordinatorNode = CalciteTests.mockCoordinatorNode();
+      return CalciteTests.mockDruidNodeDiscoveryProvider(coordinatorNode);
+    }
+  }
+
+  /**
+   * Closely related to {@link CoreInjectorBuilder#forServer()}
+   */
+  private void installForServerModules(CoreInjectorBuilder builder)
+  {
+    builder.add(
+        new Log4jShutterDownerModule(),
+        new LifecycleModule(),
+        ExtensionsModule.SecondaryModule.class,
+        new DruidAuthModule(),
+        TLSCertificateCheckerModule.class,
+        EmitterModule.class,
+        HttpClientModule.global(),
+        HttpClientModule.escalatedGlobal(),
+        new HttpClientModule("druid.broker.http", Client.class, true),
+        new HttpClientModule("druid.broker.http", EscalatedClient.class, true),
+        new CuratorModule(),
+        new AnnouncerModule(),
+        new SegmentWriteOutMediumModule(),
+        new ServerModule(),
+        new StorageNodeModule(),
+        new JettyServerModule(),
+        new ExpressionModule(),
+        new NestedDataModule(),
+        new DiscoveryModule(),
+        new ServerViewModule(),
+        new MetadataConfigModule(),
+        new DerbyMetadataStorageDruidModule(),
+        new JacksonConfigManagerModule(),
+        new CoordinatorDiscoveryModule(),
+        new LocalDataStorageDruidModule(),
+        new TombstoneDataStorageModule(),
+        new FirehoseModule(),
+        new JavaScriptModule(),
+        new AuthenticatorModule(),
+        new AuthorizerModule(),
+        new AuthorizerMapperModule(),
+        new StartupLoggingModule(),
+        new ExternalStorageAccessSecurityModule(),
+        new ServiceClientModule(),
+        new StorageConnectorModule(),
+        new SqlModule(),
+        ServerInjectorBuilder.registerNodeRoleModule(ImmutableSet.of())
+    );
+  }
+
+  /**
+   * Closely related to {@link CliBroker#getModules}.
+   */
+  static List<? extends Module> brokerModules()
```

Comment:
Where do we bind the ClientQuerySegmentWalker here? I haven't gone through the complete PR, however I wanted to know that so that I have a clearer understanding of what's going on.

Reply:
the role of this module is only to expose the `ComponentSupplier` as a broker; the `QuerySegmentWalker` is a more core part which is needed for regular tests as well - its exposed at [SqlTestFramework#TestSetupModule#querySegmentWalker](
https://github.com/apache/druid/pull/16624/files#diff-46f998221b1e961b0ba478781a64ae2816c5b034de050d4f864746981e4680caL572)

actually this is the PR I was finally able to pull that object into the guice world - doing these from guice will make it possible to get rid of the complicated way the system have launched - and could also make it easier to customize the environment further